### ISSUE TITLE
transformer_tutorial.py mentions log softmax even though it's not used.

### ISSUE THREAD

**Issue by Fraser-Greenlee** [pytorch/tutorials#800]
After this merge https://github.com/pytorch/tutorials/pull/695/files
LogSoftmax is not used even though the tutorial text says it is.
I thought the code was wrong and added the old softmax in which made it preform worse.

**Comment by Fraser-Greenlee** [comment]
It would be great to update the text & explain why a softmax is not used.

**Comment by Fraser-Greenlee** [comment]
Why the softmax was removed https://github.com/pytorch/tutorials/issues/682

**Comment by Fraser-Greenlee** [comment]
Where softmax is still mentioned https://github.com/pytorch/tutorials/blob/master/beginner_source/transformer_tutorial.py#L44
